modules/core/src/smithy4s/schema/Field.scala

```diff
@@ -17,166 +17,73 @@
-  ) extends Field[F, S, A] {
-    type T = A
-    override def toString(): String = s"Required($label, ...)"
-    override def mapK[G[_]](fk: F ~> G): Field[G, S, A] =
-      Required(label, fk(instance), get)
-    override def instanceA(onOptional: ToOptional[F]): F[A] = instance
-    override def fold[B](folder: Field.Folder[F, S, B]): B =
-      folder.onRequired(label, instance, get)
-    override def foldK[G[_]](folder: Field.FolderK[F, S, G]): G[A] =
-      folder.onRequired(label, instance, get)
-    override def leftFolder[B](folder: Field.LeftFolder[F, B]): (B, S) => B = {
-      val partiallyApplied = folder.compile(label, instance)
-      (b, s) => partiallyApplied(b, get(s))
-    }
-    override def isRequired: Boolean = true
-    override def foreachT(s: S)(f: A => Unit): Unit = f(get(s))
+  lazy val getDefaultValue: Option[A] =
+    schema.getDefaultValue
 
-    override def contramap[S0](f: S0 => S): Field[F, S0, A] =
-      Required(label, instance, f.andThen(get))
-  }
+  def isDefaultValue(a: A): Boolean = getDefaultValue.contains(a)
 
-  private final case class Optional[F[_], S, A](
-      label: String,
-      instance: F[A],
-      get: S => Option[A]
-  ) extends Field[F, S, Option[A]] {
-    type T = A
-    override def toString = s"Optional($label, ...)"
-    override def mapK[G[_]](fk: F ~> G): Field[G, S, Option[A]] =
-      Optional(label, fk(instance), get)
-    override def instanceA(onOptional: ToOptional[F]): F[Option[A]] =
-      onOptional.apply(instance)
-    override def fold[B](folder: Field.Folder[F, S, B]): B =
-      folder.onOptional(label, instance, get)
-    override def foldK[G[_]](folder: Field.FolderK[F, S, G]): G[Option[A]] =
-      folder.onOptional(label, instance, get)
-    override def leftFolder[B](folder: Field.LeftFolder[F, B]): (B, S) => B = {
-      val partiallyApplied = folder.compile(label, instance)
-      (b, s) =>
-        get(s) match {
-          case Some(a) => partiallyApplied(b, a)
-          case None    => b
-        }
+  def getIfNonDefault(s: S): Option[A] = {
+    val a = get(s)
+    getDefaultValue match {
+      case Some(`a`) => None
+      case _         => Some(a)
     }
-    override def isRequired: Boolean = false
-    override def foreachT(s: S)(f: A => Unit): Unit = get(s).foreach(f)
-
-    override def contramap[S0](f: S0 => S): Field[F, S0, Option[A]] =
-      Optional(label, instance, f.andThen(get))
-  }
-
-
-  // format: off
-  trait FolderK[F[_], S, G[_]]{
-    def onRequired[A](label: String, instance: F[A], get: S => A): G[A]
-    def onOptional[A](label: String, instance: F[A], get: S => Option[A]): G[Option[A]]
   }
 
-  trait Folder[F[_], S, B] {
-    def onRequired[A](label: String, instance: F[A], get: S => A): B
-    def onOptional[A](label: String, instance: F[A], get: S => Option[A]): B
-  }
-
-  trait LeftFolder[F[_], B] {
-    def compile[T](label: String, instance: F[T]): (B, T) => B
-  }
+  def hasDefaultValue: Boolean = getDefaultValue.isDefined
```

Comment:
this can be used to statically decide whether to invoke `get` or `getIfNonDefault` when implementing encoders. `get` is obviously more performant, as no comparison is performed, nor boxing.